scripts/clean-shrinkwrap.js

```diff
@@ -17,13 +17,28 @@
 
 const _ = require('lodash');
 const path = require('path');
-const jsonfile = require('jsonfile');
-const packageJSON = require('../package.json');
-const shrinkwrapIgnore = _.union(packageJSON.shrinkwrapIgnore, _.keys(packageJSON.optionalDependencies));
-const SHRINKWRAP_PATH = path.join(__dirname, '..', 'npm-shrinkwrap.json');
+const pkg = require('../package.json');
+const spawn = require('child_process').spawn;
+const ignore = _.union(pkg.shrinkwrapIgnore, _.keys(pkg.optionalDependencies));
 
-const shrinkwrapContents = jsonfile.readFileSync(SHRINKWRAP_PATH);
-shrinkwrapContents.dependencies = _.omit(shrinkwrapContents.dependencies, shrinkwrapIgnore);
-jsonfile.writeFileSync(SHRINKWRAP_PATH, shrinkwrapContents, {
-  spaces: 2
-});
+console.log('Removing:', ignore.join(', '));
+
+/**
+ * Run an npm command
+ * @param {Array} command - list of arguments
+ * @returns {ChildProcess}
+ */
+const npm = (command) => {
+  return spawn('npm', command, {
+    cwd: path.join(__dirname, '..'),
+    env: process.env,
+    stdio: [ process.stdin, process.stdout, process.stderr ]
+  });
+};
+
+npm([ 'rm', '--ignore-scripts' ].concat(ignore))
+  .once('close', () => {
+    npm([ 'prune' ]).once('close', () => {
+      console.log('Done.');
```

Comment:
If all this is doing is executing external commands, my gut instinct is saying that it might be easier done in a bash script? ;)

EDIT: But then again, perhaps we ought to restrict Bash scripts from being run from the `Makefile`, and (for cross-platform / self-containment reasons) any jobs run from `package.json` probably do work best as Javascript.